src/components/search/search-result.component.tsx

```diff
@@ -67,26 +68,25 @@ const SearchResultComponent: React.FC<SearchResultProps> = ({ isAnonymous, user,
 
   useEffect(() => {
     // list all transaction user id as param
-    checkFriendStatus(users.map(item => item.id));
+    if (user) checkFriendStatus([user.id]);
   }, []);
 
   const router = useRouter();
   const redirectToProfilePage = (url: string) => {
     router.push(`/${url}`);
   };
 
-  const getFriendStatus = (user: User): FriendStatus | null => {
+  const getFriendStatus = (friendUser: User): FriendStatus | null => {
```

Comment:
this is confusing because you're referring to the current user and its friend as friend

```suggestion
  const getFriendStatus = (user: User): FriendStatus | null => {
```

Reply:
actually there is 2 var user before (at props & parameter name), jadi aku coba ganti nama lain, ok untuk rename tapi mungkin bukan user lagi

Reply:
ini udah di-rename? @immhrmwn 
kalo udah gue resolve